### DISCUSSION TITLE
Storage randomly dropped

### DISCUSSION THREAD

**Discussion by scottrobertson** [community/community#5328] [Codespaces]
So, I was working on something this morning and came back an hour later. Clicked "Start" on the Codespaces, and it seems like all of the storage has been dropped. I had to run yarn and bundle again, and also I keep getting this error now when trying to perform any git actions.

![Screenshot 2021-08-27 at 13 41 14](https://user-images.githubusercontent.com/68361/131128864-cb21f834-abb8-493d-bc33-9e2bd5eed90d.png)

**Comment by Chuxel** [comment]
This shouldn't happen, so if you (or others reading this) see it again, please post the codespace URI so we can take a look at the logs and track down exactly what happened.

**Comment by Fengyi-Li** [comment]
We checked your Codespace and found that this Codespace was rebuilt at creation. A new container would be created after rebuilding. Can you show us the devcontainer config of this Codespace? It would be helpful for the investigation. Thanks!

**Reply by scottrobertson** [reply]
> We checked your Codespace and found that this Codespace was rebuilt at creation. A new container would be created after rebuilding. Can you show us the devcontainer config of this Codespace? It would be helpful for the investigation. Thanks!

I can share the devcontainer, but i cannot share the Dockerfile unfortunately. Is that still useful?

**Reply by Fengyi-Li** [reply]
Yes.

**Reply by Fengyi-Li** [reply]
You can also email it to fengyili@microsoft.com if you don't want to show it in the public thread.

**Reply by scottrobertson** [reply]
@Fengyi-Li sent :) thanks.

**Comment by jkeech** [comment]
@scottrobertson, thank you for the bug report! We've root caused this to an issue with codespaces that use Docker Compose. If the image tag (or other docker compose configuration) changes, then the subsequent `docker-compose up` on the next startup of the codespace can result in `docker-compose` automatically recreating the container. This is what happened to your codespace. Here's the high level timeline for your codespace:

1) Codespace was rebuilt into a docker-compose configuration using image with tag `X`
2) The docker compose configuration was edited at runtime (either manually, or via `git pull`, etc) to point to tag `Y`
3) The codespace was not rebuilt, so it should still be using image with tag `X`
4) The codespace stops automatically due to inactivity after some period of time
5) 🐛 When you later come back and the codespace starts up, we call `docker-compose up` which internally recreates the container, even through you didn't choose to rebuild

We are going to fix this by passing the `--no-recreate` on the `docker-compose up` command to ensure that `docker-compose` does not decide to recreate the container even if the configuration on disk changes a bit.

**Reply by jkeech** [reply]
@scottrobertson, we've pushed out a fix that adds `--no-recreate` on docker-compose codespaces when starting the codespace. You should be able to verify view searching for that flag in the creation log file after starting the codespace. Let us know if you continue to see any issues!
